src/Utils/config.py

```diff
@@ -0,0 +1,72 @@
+import os
+import yaml
+
+from dotenv import load_dotenv
+
+from common.constants import *
+
+class Config:
+    def __init__(self):
+        self.load_config()
+        self.print_config()
+        self.validate_configurations()
+
+    def load_config(self):
+        load_dotenv() # Take environment variables from .env
+        config_path = os.path.join(os.path.dirname(__file__), "../..", "config", "default_config.yaml")
+        with open(config_path, "r") as f:
+            config = yaml.safe_load(f)
+
+        # Override default configuration with any environment variables if they exist.
+        for key in config.keys():
+            env_value = os.getenv(key)
+            if env_value is not None:
+                config[key] = env_value
+
+        self.__dict__.update(config)
+
+        self.TOKENIZERS_PARALLELISM = False
+        self.LLM_API_KEY = os.getenv(LLM_API_KEY)
+        
+    def print_config(self):
+        print(" Process started! ".center(80, '-'))
+        print("".center(80, '-'))
+        for key, value in self.__dict__.items():
+            print(f"{key}={value}")
```

Comment:
Please do not print the `LLM_API_KEY` value into the terminal. It's a best practice that we should not make visible our token in logs.

`LLM_API_KEY: ***********`